gmso/tests/test_expression.py

```diff
@@ -0,0 +1,243 @@
+import pytest
+
+import sympy
+import unyt as u
+
+from gmso.utils.expression import _PotentialExpression
+from gmso.tests.base_test import BaseTest
+
+
+class TestExpression(BaseTest):
+
+    def test_expression(self):
+        expression = _PotentialExpression(
+            expression='a*x+b',
+            independent_variables='x',
+            parameters={
+                'a': 1.0 * u.dimensionless,
+                'b': 2.0 * u.dimensionless
+            }
+        )
+
+        assert expression.expression == sympy.sympify('a*x+b')
+        assert 'a' in expression.parameters.keys()
+        assert 'b' in expression.parameters.keys()
+        assert expression.parameters['a'] == 1.0 * u.dimensionless
+        assert expression.parameters['b'] == 2.0 * u.dimensionless
+
+    def test_expression_multiple_indep_vars(self):
+        expression = _PotentialExpression(
+            expression='a^2+2*a*b+b^2+2*theta*phi',
+            independent_variables={'theta', 'phi'},
+            parameters={
+                'a': 2.0 * u.nm,
+                'b': 2.0 * u.rad
+            }
+        )
+        theta = sympy.Symbol('theta')
+        phi = sympy.Symbol('phi')
+        assert theta in expression.independent_variables
+        assert phi in expression.independent_variables
+        assert theta in expression.expression.free_symbols
+        assert phi in expression.expression.free_symbols
+
+    def test_invalid_expression(self):
+        with pytest.raises(ValueError) as e:
+            expression = _PotentialExpression(
+                expression='a*x+b',
+                independent_variables='x',
+                parameters={
+                    'sigma': 1.0 * u.nm,
+                    'phi': 1.0 * u.rad
+                }
+            )
+            assert 'ValueError: Missing necessary dependencies to ' \
+                   'evaluate potential expression. Missing symbols: {b, a}' in e
+
+    def test_invalid_indep_vars(self):
+        with pytest.raises(ValueError) as e:
+            expression = _PotentialExpression(
+                expression='a*x+b',
+                independent_variables='j',
+                parameters=None
+            )
+            assert "symbol j is not in expression's free symbols Cannot " \
```

Comment:
Just a comment that I really like this check of the type of exception raised. Will be a way for me to update some of my own tests.